Framework/DataHandling/inc/MantidDataHandling/LoadGeometry.h

```diff
@@ -24,13 +30,18 @@ namespace LoadGeometry {
 
 /// Determine if the Geometry file type is IDF
 bool isIDF(const std::string &filename);
+
 /// Determine if the Geometry file type is Nexus
 bool isNexus(const std::string &filename);
+
 /// Determine if the Geometry file type is Nexus
 /// version that reuses the metadata container
 bool isNexus(const std::string &filename, const std::map<std::string, std::set<std::string>> &allEntries);
-/// List allowed file extensions for geometry
-const std::vector<std::string> validExtensions();
+
+/// Define allowed file extensions for geometry
+const std::vector<std::string> validExtensions{".xml", ".nxs", ".hdf5"};
+enum class FilenameExtensionEnum { XML, NXS, HDF5, enum_count };
+typedef EnumeratedString<FilenameExtensionEnum, &validExtensions, &compareStringsCaseInsensitive> FilenameExtension;
```

Comment:
Can these be declared inside the .cpp?  Or are they declared here so they may be included in other files?

Reply:
It's the latter. `LoadGeometry.h` is included in both `LoadEmptyInstrument.cpp` and `LoadInstrument.cpp`. Vector `validExtensions` is used in both `LoadEmptyInstrument` and `LoadInstrument` when declaring the file name property. The `enum` has to be defined next to the vector - for easier maintenance.